apps/rush-lib/src/cli/utilities/InstallManager.ts

```diff
@@ -614,8 +626,14 @@ export default class InstallManager {
     //       One possible solution would be to have the shrinkwrap include information about whether the dependency
     //       is optional or not, but it does not appear to do so. Also, this would result in strange behavior where
     //       people would have different node_modules based on their system.
+    //
+    // NOTE FOR PNPM:
+    //       PNPM does not appear to support the --no-optional flag at the moment
```

Comment:
>PNPM does not appear to support the --no-optional flag at the moment [](start = 13, length = 68)

What does PNPM do with optional dependencies?  If it has the abovementioned bug, we should add this to the list of our PNPM work items, and open a GitHub issue for the PNPM project

Reply:
opened workitem 480473 in VSTS

---
In reply to: [153378567](https://github.com/Microsoft/web-build-tools/pull/449#discussion_r153378567) [](ancestors = 153378567)